x-pack/plugins/licensing/server/plugin.test.ts

```diff
@@ -4,97 +4,265 @@
+        const license = await license$.pipe(take(1)).toPromise();
+        expect(license.isAvailable).toBe(true);
+      });
 
-    const { license$ } = await plugin.setup(coreSetup);
-    const finalLicense = await license$
-      .pipe(
-        skip(1),
-        take(1)
-      )
-      .toPromise();
+      it('observable receives updated licenses', async () => {
+        const types: LicenseType[] = ['basic', 'gold', 'platinum'];
 
-    expect(finalLicense).toBeInstanceOf(License);
-  });
+        const dataClient = elasticsearchServiceMock.createClusterClient();
+        dataClient.callAsInternalUser.mockImplementation(() =>
+          Promise.resolve({
+            license: buildRawLicense({ type: types.shift() }),
+          })
+        );
+        const coreSetup = coreMock.createSetup();
+        coreSetup.elasticsearch.dataClient$ = new BehaviorSubject(dataClient);
+
+        const { license$ } = await plugin.setup(coreSetup);
+        const [first, second, third] = await license$
+          .pipe(
+            take(3),
+            toArray()
+          )
+          .toPromise();
+
+        expect(first.type).toBe('basic');
+        expect(second.type).toBe('gold');
+        expect(third.type).toBe('platinum');
+      });
+
+      it('returns a license with error when request fails', async () => {
+        const dataClient = elasticsearchServiceMock.createClusterClient();
+        dataClient.callAsInternalUser.mockRejectedValue(new Error('test'));
+        const coreSetup = coreMock.createSetup();
+        coreSetup.elasticsearch.dataClient$ = new BehaviorSubject(dataClient);
+
+        const { license$ } = await plugin.setup(coreSetup);
+        const license = await license$.pipe(take(1)).toPromise();
+        expect(license.isAvailable).toBe(false);
+        expect(license.error).toBeDefined();
+      });
+
+      it('polling continues even if there are errors', async () => {
+        const allErrors = [new Error('reason-1'), new Error('reason-2')];
+        const errors = [...allErrors];
+
+        const dataClient = elasticsearchServiceMock.createClusterClient();
+
+        dataClient.callAsInternalUser.mockImplementation(() =>
+          errors.length > 0
+            ? Promise.reject(errors.shift())
+            : Promise.resolve({
+                license: buildRawLicense(),
+              })
+        );
+
+        const coreSetup = coreMock.createSetup();
+        coreSetup.elasticsearch.dataClient$ = new BehaviorSubject(dataClient);
+
+        const { license$ } = await plugin.setup(coreSetup);
+        const [first, second, third] = await license$
+          .pipe(
+            take(3),
+            toArray()
+          )
+          .toPromise();
+
+        expect(first.error).toBe(allErrors[0]);
+        expect(second.error).toBe(allErrors[1]);
+        expect(third.type).toBe('basic');
+      });
+
+      it('fetch license immediately without subscriptions', async () => {
```

Comment:
Can we add a similar test that verifies that logging also starts without a subscription?